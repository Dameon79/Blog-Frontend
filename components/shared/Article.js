import { Component } from 'react'
import Link from 'next/link'

class Article extends Component {
  render () {
    return (
      <Link href='/article'>
        <article className='c-articles-container__article'>
          <h3 className='c-articles-container__article-title'>Dummy Article</h3>
          <img className='c-articles-container__article-image' src='/static/images/Classes.jpg'></img>
        </article>
      </Link>
    );
  }
}

export default Article