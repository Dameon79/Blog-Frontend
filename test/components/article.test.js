import Article from '../../components/shared/Article'

describe('<Article />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <Article />
    )

    expect(component).toMatchSnapshot()
  })
})