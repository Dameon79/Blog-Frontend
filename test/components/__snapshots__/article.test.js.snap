// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Article />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <article
        className="c-articles-container__article"
      >
        <h3
          className="c-articles-container__article-title"
        >
          Dummy Article
        </h3>
        <img
          className="c-articles-container__article-image"
          src="/static/images/Classes.jpg"
        />
      </article>,
      "href": "/article",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="c-articles-container__article-title"
          >
            Dummy Article
          </h3>,
          <img
            className="c-articles-container__article-image"
            src="/static/images/Classes.jpg"
          />,
        ],
        "className": "c-articles-container__article",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Dummy Article",
            "className": "c-articles-container__article-title",
          },
          "ref": null,
          "rendered": "Dummy Article",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-articles-container__article-image",
            "src": "/static/images/Classes.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "article",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <article
          className="c-articles-container__article"
        >
          <h3
            className="c-articles-container__article-title"
          >
            Dummy Article
          </h3>
          <img
            className="c-articles-container__article-image"
            src="/static/images/Classes.jpg"
          />
        </article>,
        "href": "/article",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="c-articles-container__article-title"
            >
              Dummy Article
            </h3>,
            <img
              className="c-articles-container__article-image"
              src="/static/images/Classes.jpg"
            />,
          ],
          "className": "c-articles-container__article",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Dummy Article",
              "className": "c-articles-container__article-title",
            },
            "ref": null,
            "rendered": "Dummy Article",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "c-articles-container__article-image",
              "src": "/static/images/Classes.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "article",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
